scargo/commands/check.py

```diff
@@ -295,25 +295,42 @@ def fix_file(self, file_path: Path) -> None:
 
 class ClangTidyChecker(CheckerFixer):
     check_name = "clang-tidy"
-    build_path = Path("./build/")
+    build_path: Optional[Path] = None
 
     def check_file(self, file_path: Path) -> CheckResult:
-        cmd = ["clang-tidy", str(file_path)]
+        try:
+            bf_list = os.listdir(Path(self._config.project_root, "build"))
+        except:
+            logger.error("Build folder does not exist.")
+            logger.info("Did you run `scargo build`?")
+            sys.exit(1)
+
+        for bf in bf_list:
+            if bf in self._config.profiles.keys():
+                self.build_path = Path(self._config.project_root, "build", bf)
+                break
+
+        if not self.build_path:
+            logger.error(
+                "Build folder for any of supported build types does not exist."
+            )
+            logger.info("Did you run `scargo build`?")
+            sys.exit(1)
+
+        cmd: List[str]
+        # Check if compilation database exists:
+        if not Path(self.build_path, "compile_commands.json").exists():
+            logger.error("Compilation database does not exist.")
+            logger.info("Did you run `scargo build`?")
+            sys.exit(1)
+
         if self._config.project.target.family == "esp32":
-            cmd.extend(["-p", str(self.build_path)])
-            if not Path(self.build_path, "compile_commands.json").exists():
-                # creates compilation database and runs run-clang-tidy.py on the whole project
-                # (the latter is not needed, but there's no option to suppress it)
-                subprocess.run(
-                    ["idf.py", "clang-check"],
-                    stdout=subprocess.DEVNULL,
-                    stderr=subprocess.DEVNULL,
-                    check=False,
-                )
-        if file_path.suffix == ".h":
-            cmd.extend(["--", "-x", "c++"])
-        if "--" not in cmd and self._config.project.target.family != "esp32":
-            cmd.append("--")
+            cmd = self.__get_cmd_esp32(file_path)
+        elif self._config.project.target.family == "stm32":
+            cmd = self.__get_cmd_stm32(file_path)
+        else:
```

Comment:
Wouldn't be better to add condition check (== 'x86') to avoid bug in future in case of adding new targets?

Reply:
Changed